src/backend.rs

```diff
@@ -687,12 +687,10 @@ impl DatabaseRef for SharedBackend {
         })
     }
 
-    fn block_hash_ref(&self, number: U256) -> Result<B256, Self::Error> {
-        if number > U256::from(u64::MAX) {
+    fn block_hash_ref(&self, number: u64) -> Result<B256, Self::Error> {
+        if number > u64::MAX {
             return Ok(KECCAK_EMPTY);
         }
```

Comment:
this check is now redundant and can be removed entirely

Reply:
Completely missed that
Fixed